File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/klaytnDebug/logging/DebugVerbosityByIDTest.java
package opensdk.sdk.apis.klaytnDebug.logging;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.DebugVerbosityByIDResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;

@DisplayName("Debug RPC Test")
public class DebugVerbosityByIDTest {
    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC debug_verbosityById")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        int id = 1;
        int level = 3;

        DebugVerbosityByIDResponse response = sdk.debug.verbosityByID(id, level).send();

        assertNotNull(response);
        assertNull(response.getError());
    }
}

File: web3rpc/sdk/client/javascript/openapi-test/test/debug/logging/DebugVerbosityByID.test.js
const OpenSdk = require("@klaytn/web3rpc");
const { expect } = require("@jest/globals");
const { RPC } = require("../../constant");

const sdk = new OpenSdk.DebugApi(new OpenSdk.ApiClient(RPC));

describe('debug_verbosityByID API', () => {
    test('should return debug_verbosityByID', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data).toBeDefined();
            expect(data).toBe(null);
            done();
        };

        const id = 1;
        const level = 3;

        sdk.verbosityByID(id, level, {}, callbackOne);
    });
});

